### PR TITLE
EZP-29979: As a developer I would like to use getLocation method on ContentCreateSuccessView and ContentEditSuccessView

### PR DESCRIPTION
**Pull request by kmadejski**
JIRA: https://jira.ez.no/browse/EZP-29979

### CODE REVIEW

File: lib/Content/View/ContentCreateSuccessView.php [hunk 1]
```diff
@@ -25,4 +30,20 @@ public function __construct(Response $response)
         $this->setResponse($response);
         $this->setControllerReference(new ControllerReference('ez_content_edit:createWithoutDraftSuccessAction'));
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location|null $location
+     */
+    public function setLocation(?Location $location)
```
Comment (ViniTou) [inline lib/Content/View/ContentCreateSuccessView.php]: Not sure if there is any case when this is set to null

Comment (kmadejski) [reply lib/Content/View/ContentCreateSuccessView.php]: Yep, just wanted to be consistent with https://github.com/ezsystems/repository-forms/blob/master/lib/Content/View/ContentEditView.php#L61, but it seems that indeed there is no case for that. Let's wait for @webhdx 🙂

Comment (andrerom) [reply lib/Content/View/ContentCreateSuccessView.php]: Only hypothetical case would be when you want to remove location.

For getLocation it's needed for cases where there are no locations _(most notably draft, but also technically possible with location less content)_, but I also don't see a strong need to support this on setLocation().

File: lib/Content/View/ContentCreateSuccessView.php [hunk 2]
```diff
@@ -25,4 +30,20 @@ public function __construct(Response $response)
         $this->setResponse($response);
         $this->setControllerReference(new ControllerReference('ez_content_edit:createWithoutDraftSuccessAction'));
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     */
+    public function setLocation(Location $location)
```
Comment (webhdx) [inline lib/Content/View/ContentCreateSuccessView.php]:
```suggestion
    public function setLocation(Location $location): void
```

File: lib/Content/View/ContentEditSuccessView.php [hunk 3]
```diff
@@ -25,4 +30,20 @@ public function __construct(Response $response)
         $this->setResponse($response);
         $this->setControllerReference(new ControllerReference('ez_content_edit:editVersionDraftSuccessAction'));
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     */
+    public function setLocation(Location $location)
```
Comment (webhdx) [inline lib/Content/View/ContentEditSuccessView.php]:
```suggestion
    public function setLocation(Location $location): void
```

**Comment by kmadejski** [comment]
@andrerom / @ViniTou I have to allow passing `null` as `$location` due to the issue found by QA. 

@mnocon could you please retest?

**Comment by kmadejski** [comment]
@mnocon do you see any chance to give it another round of testing? 🙂

**Comment by mnocon** [comment]
@kmadejski there is a chance 😉 But you're probably aware that 60% of our QA team is on sick leave and it complicates things a bit

**Comment by kmadejski** [comment]
Good question @mnocon, honestly, I don't know. @ViniTou / @webhdx / @Nattfarinn / @andrerom 
 WDYT?

**Pull request merged**